### DISCUSSION TITLE
Help

### DISCUSSION THREAD

**Discussion by cyon30** [kellerza/sunsynk#82] [General]
I realy need help. I want to pull all the sensors but I don't understand where to past the .py

**Comment by kellerza** [comment]
Did you read https://github.com/kellerza/sunsynk ?

No need to discuss .py at all

**Comment by cyon30** [comment]
I have, I running Proxmox and CT - Home Assistan on. I have the RS485 in and I can pull the totals from the sunsynk. But I want more sensors. I get lost with the github in HA.

**Reply by kellerza** [reply]
Not sure I follow, but if the addon is working, you simply add more sensors to the list in the configuration, under `SENSORS:` 

You can find the names of all the sensors in `definitions.py` as per the addon documentation.

**Comment by duanemck** [comment]
@cyon30 This is my config yml if it helps?

```
PORT: tcp://[mbusd-host-name]:502
DEVICE: /dev/ttyAMA0
DRIVER: umodbus
SUNSYNK_ID: "MY_SERIAL"
SENSOR_PREFIX: sunsynk
SENSORS:
  - total_pv_energy
  - total_battery_charge
  - total_battery_discharge
  - total_grid_export
  - total_grid_import
  - battery_soc
  - essential_power
  - non_essential_power
  - grid_ct_power
  - battery_power
  - inverter_power
  - pv1_power
  - pv2_power
  - aux_power
  - grid_connected_status
  - prog1_time
  - prog1_power
  - prog1_capacity
  - prog1_charge
  - prog1_voltage
  - prog2_time
  - prog2_power
  - prog2_capacity
  - prog2_charge
  - prog2_voltage
  - prog3_time
  - prog3_power
  - prog3_capacity
  - prog3_charge
  - prog3_voltage
  - prog4_time
  - prog4_power
  - prog4_capacity
  - prog4_charge
  - prog4_voltage
  - prog5_time
  - prog5_power
  - prog5_capacity
  - prog5_charge
  - prog5_voltage
  - prog6_time
  - prog6_power
  - prog6_capacity
  - prog6_charge
  - prog6_voltage
READ_SENSORS_BATCH_SIZE: 60
NUMBER_ENTITY_MODE: auto
PROFILES:
  - system_mode
MODBUS_SERVER_ID: 1
MQTT_HOST: 192.168.1.1
MQTT_PORT: 1883
MQTT_USERNAME: username
MQTT_PASSWORD: password
DEBUG: 0
```

**Reply by cyon30** [reply]
> @cyon30 This is my config yml if it helps?
> 
> ```
> PORT: tcp://[mbusd-host-name]:502
> DEVICE: /dev/ttyAMA0
> DRIVER: umodbus
> SUNSYNK_ID: "MY_SERIAL"
> SENSOR_PREFIX: sunsynk
> SENSORS:
>   - total_pv_energy
>   - total_battery_charge
>   - total_battery_discharge
>   - total_grid_export
>   - total_grid_import
>   - battery_soc
>   - essential_power
>   - non_essential_power
>   - grid_ct_power
>   - battery_power
>   - inverter_power
>   - pv1_power
>   - pv2_power
>   - aux_power
>   - grid_connected_status
>   - prog1_time
>   - prog1_power
>   - prog1_capacity
>   - prog1_charge
>   - prog1_voltage
>   - prog2_time
>   - prog2_power
>   - prog2_capacity
>   - prog2_charge
>   - prog2_voltage
>   - prog3_time
>   - prog3_power
>   - prog3_capacity
>   - prog3_charge
>   - prog3_voltage
>   - prog4_time
>   - prog4_power
>   - prog4_capacity
>   - prog4_charge
>   - prog4_voltage
>   - prog5_time
>   - prog5_power
>   - prog5_capacity
>   - prog5_charge
>   - prog5_voltage
>   - prog6_time
>   - prog6_power
>   - prog6_capacity
>   - prog6_charge
>   - prog6_voltage
> READ_SENSORS_BATCH_SIZE: 60
> NUMBER_ENTITY_MODE: auto
> PROFILES:
>   - system_mode
> MODBUS_SERVER_ID: 1
> MQTT_HOST: 192.168.1.1
> MQTT_PORT: 1883
> MQTT_USERNAME: username
> MQTT_PASSWORD: password
> DEBUG: 0
> ```

Wow.. Thank you so much... going on my knees here... Thank you

**Comment by cyon30** [comment]
What do you do with all the programs?

**Reply by kellerza** [reply]
You can view & change your inverter settings in the UI

![image](https://user-images.githubusercontent.com/6756881/212750870-b5f26859-94bb-4206-aa12-86c8a4e249d9.png)

**Comment by cyon30** [comment]
Wow awesome. got mine working. 

So I got something to work. I custom my list and added the temp from the inverter and now my google home can see the temp and programs but I would love to get the other sensors to be intergrated so I can ask google like the "soc" or "PV Power" or "grid connection"
How can I add more? 

![image](https://user-images.githubusercontent.com/20164240/212831227-b0cb6fcb-8b6d-488c-9a71-5689bc0caf61.png)

![image](https://user-images.githubusercontent.com/20164240/212830970-7431a989-ca9a-48d8-b13e-bf60130e38c8.png)

**Comment by cyon30** [comment]
Hi, 
Can someone help. If I go and the add temperture sensor my google assistain see the sensor. But I would like to have the SOC and more.. 
Like to have google to beable to give me the status of the sunsynk. 
What must I do?
![image](https://user-images.githubusercontent.com/20164240/213182989-c7e9baf3-d5ec-45d5-8a1e-2ec8898f37e2.png)

**Reply by kellerza** [reply]
Hi @cyon30 ... the Addon documentation takes you to the definitions, but let me use a concrete example

If you look at [this line](https://github.com/kellerza/sunsynk/blob/main/sunsynk/definitions.py#L30)
```python
Sensor(184, "Battery SOC", "%"),
```

now you can add this to your configuration, under `SENSORS:` in the format you like. While most people uses the first format, even the second and the third format should be accepted... 

```yaml
SENSORS:
  - battery_soc
  - battery soc
  - Battery SOC
```

As a test add all three above and then see if the log complains that you should only add things once 

Now you can do this for any sensor in the definitions.py` file. If you look at @duanemck 's config you will find all his sensors in that file!

Just stay away from the last 20 lines, they show deprecated names of sensors. The reason is we now have better, or more consistent names, for the same sensors!

**Reply by kellerza** [reply]
https://kellerza.github.io/sunsynk/reference/definitions#three-phase-inverter-sensor-definitions
